akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1359,6 +1378,32 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def recoverWith(pf: PartialFunction[Throwable, _ <: Graph[SourceShape[Out], NotUsed]]): javadsl.Flow[In, Out, Mat] =
     new Flow(delegate.recoverWith(pf))
 
+  /**
+   * RecoverWith allows to switch to alternative Source on flow failure. It will stay in effect after
+   * a failure has been recovered so that each time there is a failure it is fed into the `pf` and a new
+   * Source may be materialized.
+   *
+   * Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements.
+   * This stage can recover the failure signal, but not the skipped elements, which will be dropped.
+   *
+   * Throwing an exception inside `recoverWith` _will_ be logged on ERROR level automatically.
+   *
+   * '''Emits when''' element is available from the upstream or upstream is failed and element is available
+   * from alternative Source
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes or upstream failed with exception pf can handle
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   */
+  @deprecated("Use recoverWithRetries instead.", "2.4.4")
+  def recoverWith(clazz: Class[_ <: Throwable], graph: Graph[SourceShape[Out], NotUsed]): javadsl.Flow[In, Out, Mat] =
+    recoverWith {
+      case elem if clazz.isInstance(elem) ⇒ graph
+    }
```

Comment:
I think we should remove the deprecation. Also, note that this signature is not correct. In the Scala one it is a `PartialFunction[Throwable, Source]` which means this should not accept a `Graph`, but instead a `Function[Throwable, Source]`. 

Would also be good if you could doublecheck that this type-bound actually does what you'd expect on the Java side, I don't feel 100% sure about that.

Reply:
good catch. I think it should be a `Supplier`, since the purpose is to do the matching with `clazz` only.

Regarding `Source` vs `Graph[SourceShape]`? I'm not sure, we use `Graph[SourceShape]` in other places? Doesn't that work?

Reply:
Ah, yes. `Supplier` is fine. `Source` was just because I was too lazy to write out the full `Graph[SourceShape[...`, I agree!